### ISSUE TITLE
NullReferenceException at OpenRA.Network.UPnP.ForwardPort(Int32 lifetime)

### ISSUE THREAD

**Issue by drkingpo** [OpenRA/OpenRA#11346]
Hi there,

I'm trying to get the Dedicated Server running on Windows 7 x64, but running directling in a crash:

-----------------------------------------

C:\Program Files (x86)\OpenRA>set Name="KPO3 Dedicated Server"
C:\Program Files (x86)\OpenRA>set Mod=ra
C:\Program Files (x86)\OpenRA>set Dedicated=True
C:\Program Files (x86)\OpenRA>set DedicatedLoop=True
C:\Program Files (x86)\OpenRA>set ListenPort=1234
C:\Program Files (x86)\OpenRA>set ExternalPort=1234
C:\Program Files (x86)\OpenRA>set AdvertiseOnline=True
C:\Program Files (x86)\OpenRA>set Map=Synergy
C:\Program Files (x86)\OpenRA>set Password=

C:\Program Files (x86)\OpenRA>OpenRA.Server.exe Game.Mod=ra Server.Dedicated=Tru
e Server.DedicatedLoop=True Server.Name="KPO3 Dedicated Server" Server.ListenPor
t=1234 Server.ExternalPort=1234 Server.AdvertiseOnline=True Server.Map=Synergy S
erver.Password=
[2016-05-23T14:05:43] Starting dedicated server for mod: ra

Unhandled Exception: System.NullReferenceException: Object reference not set to
an instance of an object.
   at OpenRA.Network.UPnP.ForwardPort(Int32 lifetime)
   at OpenRA.Server.Server..ctor(IPEndPoint endpoint, ServerSettings settings, M
odData modData, Boolean dedicated)
   at OpenRA.Server.Program.Main(String[] args)

-----------------------------------------

dedicated-debug and dedicated-server.log are emtpy.
in dedicated-perf.log im find the following:
    54 ms a-path-beyond.oramap
     3 ms alaska-anarchy-redux.oramap
     2 ms all-connected.oramap
     3 ms arctic-triangle-affair.oramap
    14 ms asymmetric-battle.oramap
     1 ms bad-neighbors.oramap
     2 ms barracuda.oramap
     2 ms behind-the-veil.oramap
     3 ms blitzkrieg.oramap
     8 ms bloody-delta.oramap
     2 ms bombardment-islands.oramap
     2 ms caffeinated.oramap
     2 ms calm-before-storm.oramap
     3 ms chaos-canyon.oramap
     2 ms chokepoint.oramap
     9 ms coastal-influence.oramap
     3 ms cold-front-1v1.oramap
     2 ms cold-front.oramap
     3 ms doubles.oramap
     3 ms doughnut.oramap
     1 ms east-vs-west.oramap
     6 ms encounter.oramap
     3 ms engagement.oramap
     2 ms equal-opportunity.oramap
     2 ms first-come-first-served.oramap
     2 ms forest-path.oramap
     2 ms ghost-town.oramap
     6 ms haos-ridges.oramap
     3 ms high-and-low.oramap
    11 ms keep-off-the-grass-2.oramap
     2 ms mad-science.oramap
     1 ms man-to-man.oramap
     2 ms marooned-2.oramap
     3 ms mass-confliction.oramap
     6 ms north-by-northwest.oramap
     2 ms northwestpassage.oramap
     3 ms ore-lord.oramap
     3 ms pool-party.oramap
     2 ms pressure.oramap
     2 ms puddles-redux.oramap
     6 ms regeneration-basin.oramap
     2 ms Sahara.oramap
    14 ms seaside-2.oramap
     2 ms sidestep.oramap
     2 ms singles.oramap
     3 ms snowy-island.oramap
     1 ms snowy-ridge.oramap
     2 ms sudden-death.oramap
     5 ms suffrage.oramap
     2 ms sunstroke.oramap
     2 ms synergy.oramap
     2 ms tabula-rasa.oramap
     3 ms tainted-peak.oramap
     3 ms temperal.oramap
     6 ms tournament-island.oramap
     3 ms vegetation.oramap
     1 ms warwind.oramap
     2 ms winter-storm.oramap
     4 ms allies-01
     3 ms allies-02
     3 ms allies-03a
    16 ms allies-03b
     4 ms allies-05a
     2 ms bomber-john
     2 ms center-of-attention-redux-2
     3 ms desert-shellmap
     1 ms drop-zone
     2 ms drop-zone-battle-of-tikiaki
     9 ms drop-zone-w
     3 ms fort-lonestar
     5 ms intervention
     3 ms koth-hopes-anchor
     5 ms monster-tank-madness
     3 ms poland-raid
     3 ms snow town
    10 ms soviet-01
     3 ms soviet-02a
     3 ms soviet-02b
     4 ms soviet-03
    15 ms soviet-04a
     3 ms soviet-04b
     4 ms soviet-05
     7 ms soviet-06a
     4 ms soviet-06b
     3 ms soviet-07
     3 ms survival01
     3 ms survival02
     2 ms training-camp
-----------------------------------------------------------
I'm getting the same on another Windows 8 x64 PC.  Both PCs are VM's, so I'm not able to start the Game itself.
But I'm getting the Same issue on two systems here at my home. One Win8 and one Win10.
Both are able  to run the OpenRA GUI Game itself and also to Host Games and than and play, but getting the same issues if I try to set up the Server.
I'm using the latest - release-20160508 - .

Anything else you need?
Many thanks and regards
KingPo

**Comment by Mailaender** [comment]
Try starting the server with the command line parameter `Server.DiscoverNatDevices=False`

**Comment by Mailaender** [comment]
https://github.com/OpenRA/OpenRA/pull/11286 will overhaul our UPnP feature completely so I don't want to investigate this problem any deeper.

**Comment by drkingpo** [comment]
Hi there,

thx I added "Server.DiscoverNatDevices=False" to the parameters, but runs into the same issue.

Additionally to crosstest I installed Ubuntu on the different system, and it working.... just to check my Hardware & Configuration is fine so far.
Curiosily I dont get the Password protection working with the Ubuntu Dedicated Server. :(

But If you rework the UPnP feature I'm looking forward to see an solution later for Windows.

Thanks all for your work.
